### DISCUSSION TITLE
Correlation between docker image tags and releases

### DISCUSSION THREAD

**Discussion by bmoqimi** [quay/quay#879] [Q&A]
Hi everyone,

We have been trying to migrate from the redhat paid licenses to the open source project but I am very confused with the release notes and versioning. Can anyone shed some light on it please? 
What is the relationship between releases in these three links: 
https://docs.projectquay.io/release_notes.html
https://github.com/quay/quay/releases
https://quay.io/repository/projectquay/quay?tab=tags 

Thanks a lot :-) 🙇

**Comment by BillDett** [comment]
Our upstream release process has been evolving over the last few releases as we moved away from named tags to align with the RH Quay release versions.  Right now these upstream releases are cut somewhat manually and we haven't had much bandwidth to get the automation in place to do it regularly.  The 3.6.0 alpha images are very early builds of the next minor version of Quay due out shortly- would stay clear of those- if you need something stable grab 3.5.2.

**Comment by bmoqimi** [comment]
perfect. Thanks a lot for the info.
Can you predict what would happen if we suddenly switch the image from RH 3.5.5 to projectquay 3.5.2?
or should we just wait until the 3.6 stable is out and then switch to that? 
Any hints is appreciated :-)

**Comment by BillDett** [comment]
You should be able to interchange 3.5.5 and 3.5.2 images without 
problems (aside from missing bug fixes).  We would recommend moving to 
3.6.0 specifically to pick up new version of Clair and improved operator 
support if you're running on OpenShift.  A couple new registry features 
in 3.6.0 as well (nested repos, custom OCI media type support).

On 9/1/21 10:24 AM, Babak Moghimi wrote:
>
> perfect. Thanks a lot for the info.
> Can you predict what would happen if we suddenly switch the image from 
> RH 3.5.5 to projectquay 3.5.2?
> or should we just wait until the 3.6 stable is out and then switch to 
> that?
> Any hints is appreciated :-)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/quay/quay/discussions/879#discussioncomment-1265620>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AABQYQSDCRWK5Q6LUVD7TM3T7YZR5ANCNFSM5DGB6B4A>.
> Triage notifications on the go with GitHub Mobile for iOS 
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> 
> or Android 
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
>
>
-- 
Bill Dettelback
Senior Engineering Manager - Red Hat Quay, OpenShift Image Registry
M: 732-991-0052
https://www.redhat.com/en/technologies/cloud-computing/quay
Red Hat respects your work life balance. Therefore there is no need to answer this email out of your office hours.

**Comment by inka** [comment]
Hello,

just seen that there is now 3.6.1 released here (what you call the derivate)
https://docs.projectquay.io/release_notes.html#rn-3-601

but here it is v3.6.0-alpha.9 (what you call the upstream)
https://github.com/quay/quay/releases/tag/v3.6.0-alpha.9

Do have any plans to align these release cycles? In my understanding it makes sense to release upstream first and then add the commercial flavor on top.

Anything people could help with to achieve this?

Thanks,
Ingo

**Reply by flavianmissi** [reply]
Hi @inka, the plan is indeed to release upstream first, but as Bill said other priorities are currently winning.
For now we have "unstable" builds of 3.6.z. The unstable tags are rebuilt on every push to a redhat-x.y branch, so they're the closest you can currently get to a release.

**Reply by inka** [reply]
Yeah, release 3.6.2 arrived! Thank you so much.

I suggest to close this issue then.

**Reply by flavianmissi** [reply]
Please note that for now, our Github releases only have source code artifacts associated to them. We'll be releasing container images along with them in the foreseeable future.

**Reply by bmoqimi** [reply]
The appearance of the v3.6.2 tag here is something we can rely on though I guess? https://quay.io/repository/projectquay/quay?tab=tags
That is what our images are based on (at least until the open tickets with the bugs are resolved and we don't need to patch them).

**Reply by flavianmissi** [reply]
Yes, as long as we're consistently tagging releases via git you should be able to rely on the equivalent image tags on quay.io.
Just beware that we might update the contents of the tags, since we're currently not creating RC tags.
